packages/client-sdk-nodejs/src/internal/grpc/idle-grpc-client-wrapper.ts

```diff
@@ -48,32 +53,60 @@ export class IdleGrpcClientWrapper<T extends CloseableGrpcClient>
   }
 
   getClient(): T {
-    this.logger.trace(
-      `Checking to see if client has been idle for more than ${this.maxIdleMillis} ms`
-    );
-    if (Date.now() - this.lastAccessTime > this.maxIdleMillis) {
-      this.logger.info(
-        `Client has been idle for more than ${this.maxIdleMillis} ms; reconnecting.`
+    const now = Date.now();
+
+    // Reconnect if channel is in a bad state.
+    // Although the generic type `T` only extends `CloseableGrpcClient` (which doesn't define `getChannel()`),
+    // we know that in practice, the client returned by `clientFactoryFn()` is a gRPC client that inherits from
+    // `grpc.Client`: https://grpc.github.io/grpc/node/grpc.Client.html
+    //
+    // Since `grpc.Client` defines a public `getChannel()` method, we can safely assume it's present on the client.
+    // To make TypeScript accept this, we cast the client to `unknown` first and then to `GrpcClientWithChannel`.
+    // This double-cast is necessary to override structural type checking and is safe in our controlled use case.
+    const clientWithChannel = this.client as unknown as GrpcClientWithChannel;
```

Comment:
Can we just add `getChannel` to the `CloseableGrpcClient` interface? It's an internal interface so not a breaking change (see other comment).